packages/material-ui/src/ButtonBase/TouchRipple.js

```diff
@@ -184,8 +184,8 @@ const TouchRipple = React.forwardRef(function TouchRipple(props, ref) {
         rippleX = Math.round(rect.width / 2);
         rippleY = Math.round(rect.height / 2);
       } else {
-        const clientX = event.clientX ? event.clientX : event.touches[0].clientX;
-        const clientY = event.clientY ? event.clientY : event.touches[0].clientY;
+        const clientX = event.clientX >= 0 ? event.clientX : event.touches[0].clientX;
+        const clientY = event.clientY >= 0 ? event.clientY : event.touches[0].clientY;
```

Comment:
That looks fine to me. 

It's not clear why we branched to `event.touches` if `event.clientX === 0`. Why would `event.clientX === 0` use a different path? @oliviertassinari

Reply:
I don't know, I believe the logic hasn't changed for 4 years. Maybe?

```diff
diff --git a/packages/material-ui/src/ButtonBase/TouchRipple.js b/packages/material-ui/src/ButtonBase/TouchRipple.js
index a3cd4b0db..a5febe759 100644
--- a/packages/material-ui/src/ButtonBase/TouchRipple.js
+++ b/packages/material-ui/src/ButtonBase/TouchRipple.js
@@ -184,8 +184,7 @@ const TouchRipple = React.forwardRef(function TouchRipple(props, ref) {
         rippleX = Math.round(rect.width / 2);
         rippleY = Math.round(rect.height / 2);
       } else {
-        const clientX = event.clientX ? event.clientX : event.touches[0].clientX;
-        const clientY = event.clientY ? event.clientY : event.touches[0].clientY;
+        const { clientX, clientY } = event.touches ? event.touches[0] : event;
         rippleX = Math.round(clientX - rect.left);
         rippleY = Math.round(clientY - rect.top);
       }
```

Reply:
I guess this is the usual shortcut :smile: that optimized for writability. Your new approach is even better. Because with this outdated approach I would ask what happens if `event.clientX` is smaller than 0 :wink:

Reply:
Can client x be smaller than 0? I have no idea 😆

Reply:
I'm happy we might have found a solution to the Sentry reports on #12362 :D. I hate when a third-party dependency creates exceptions, it feels like its poorly written. I had a personal poor experience with https://www.drift.com/.